scenario-compiler/src/main/scala/pl/touk/nussknacker/engine/extension/AllowedCastParametersClasses.scala

```diff
@@ -0,0 +1,29 @@
+package pl.touk.nussknacker.engine.extension
+
+import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
+import pl.touk.nussknacker.engine.definition.clazz.ClassDefinitionSet
+
+import scala.util.Try
+
+final case class AllowedCastParametersClasses(namesWithTyping: Map[String, TypingResult]) {
+  def get(className: String): Option[TypingResult] =
+    namesWithTyping.get(className)
+}
+
+object AllowedCastParametersClasses {
+
+  def apply(set: ClassDefinitionSet): AllowedCastParametersClasses =
+    new AllowedCastParametersClasses(
+      namesWithTyping = set.classDefinitionsMap
+        .map { case (clazz, classDefinition) =>
+          clazz.getName -> Try(classDefinition.clazzName).toOption
```

Comment:
Why this is wrapped in Try?

Reply:
What is done? I was only asking :)  + I see that it sill exists. `clazzName` is a field, that it won't throw an exception.